tests/network/localnet/conftest.py

```diff
@@ -38,81 +46,198 @@ def nncp_localnet() -> Generator[libnncp.NodeNetworkConfigurationPolicy]:
         yield nncp
 
 
-@pytest.fixture(scope="module")
+@pytest.fixture(scope="class")
 def namespace_localnet_1() -> Generator[Namespace]:
     yield from create_ns(name="test-localnet-ns1")  # type: ignore
 
 
-@pytest.fixture(scope="module")
+@pytest.fixture(scope="class")
 def namespace_localnet_2() -> Generator[Namespace]:
     yield from create_ns(name="test-localnet-ns2")  # type: ignore
 
 
-@pytest.fixture(scope="module")
+@pytest.fixture(scope="class")
 def vlan_id(vlan_index_number: Generator[int]) -> int:
     return next(vlan_index_number)
 
 
-@pytest.fixture(scope="module")
+@pytest.fixture(scope="class")
 def nad_localnet_1(
     namespace_localnet_1: Namespace, vlan_id: int
 ) -> Generator[netattachdef.NetworkAttachmentDefinition]:
-    with localnet_nad(namespace=namespace_localnet_1.name, name="test-localnet-nad1", vlan_id=vlan_id) as nad:
+    with localnet_nad(
+        namespace=namespace_localnet_1.name,
+        name="test-localnet-nad1",
+        vlan_id=vlan_id,
+        network_name=LOCALNET_BR_EX_NETWORK,
+    ) as nad:
         yield nad
 
 
-@pytest.fixture(scope="module")
+@pytest.fixture(scope="class")
 def nad_localnet_2(
     namespace_localnet_2: Namespace, vlan_id: int
 ) -> Generator[netattachdef.NetworkAttachmentDefinition]:
-    with localnet_nad(namespace=namespace_localnet_2.name, name="test-localnet-nad2", vlan_id=vlan_id) as nad:
+    with localnet_nad(
+        namespace=namespace_localnet_2.name,
+        name="test-localnet-nad2",
+        vlan_id=vlan_id,
+        network_name=LOCALNET_BR_EX_NETWORK,
+    ) as nad:
         yield nad
 
 
-@pytest.fixture(scope="module")
-def vm_localnet_1(nad_localnet_1: netattachdef.NetworkAttachmentDefinition) -> Generator[BaseVirtualMachine]:
+@pytest.fixture(scope="class")
+def ipv4_localnet_address_pool() -> Generator[str]:
```

Comment:
```suggestion
def ipv4_localnet_address_pool() -> str:
```
`return`  is used

Reply:
@rnetser , it does return a generator. See usage.

Reply:
missed that